kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt

```diff
@@ -1,72 +1,1085 @@
+          "Numbers bigger than itself" {
+            assertAll(nonMinNorMaxValueDoubles) {
+              it shouldNotMatchGreaterThanOrEqual it.slightlyGreater()
+              it shouldNotMatchGreaterThanOrEqual it.muchGreater()
+              
+            }
+          }
+          
+          "Positive Infinity" {
+            assertAll(nonMinNorMaxValueDoubles) {
+              it shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+          }
+          
+          "NaN" {
+            assertAll(nonMinNorMaxValueDoubles) {
+              it shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+        
+      }
+      
+      "The non-numeric double" - {
+        "NaN" - {
+          "Should not be greater than or equal to" - {
+            "Itself" {
+              NaN shouldNotMatchGreaterThanOrEqual NaN
+            }
+            
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                NaN shouldNotMatchGreaterThanOrEqual it
+              }
+            }
+            
+            "Positive Infinity" {
+              NaN shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "Negative Infinity" {
+              NaN shouldNotMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+          }
         }
+        
+        "Positive Infinity" - {
+          "Should be greater than or equal to" - {
+            "Itself" {
+              POSITIVE_INFINITY shouldMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "Negative Infinity" {
+              POSITIVE_INFINITY shouldMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+            
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                POSITIVE_INFINITY shouldMatchGreaterThanOrEqual it
+              }
+            }
+          }
+          
+          "Should not be greater than or equal to" - {
+            "NaN" {
+              POSITIVE_INFINITY shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+        
+        "Negative Infinity" - {
+          "Should be greater than or equal to" - {
+            "Itself" {
+              NEGATIVE_INFINITY shouldMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+          }
+          
+          "Should not be greater than or equal to" - {
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual it
+              }
+            }
+            
+            "Positive Infinity" {
+              NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "NaN" {
+              NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+      }
+    }
+    
+    
+  }
+  
+  private fun Double.smallDelta(): Double {
```

Comment:
We're getting there! These extensions work in more cases, but still fail for values near the representable limits. For example, `smallDelta` and the `slighty*` extensions fail because `Double.MIN_VALUE * 0.01 == Double.MIN_VALUE`. You're looking for the smallest number you can add to a double that will change its value. The term for this is [Unit of Least Precision (ULP)](https://en.wikipedia.org/wiki/Unit_in_the_last_place). The JDK has a built in function for this: [`Math.ulp`](https://docs.oracle.com/javase/10/docs/api/java/lang/Math.html#ulp(double)).

For the `muchGreater` and `muchSmaller` extensions, many large double values will end up as an infinity (e.g. `1.5e308 * 1.25 == POSITIVE_INFINITY`), and many small values will produce incorrect results (e.g. `3e-324.muchGreater() == Double.MIN_VALUE`, which is _smaller_ than `3e-324`). I suggest using a multiple of the ULP like `this + 2 * Math.ulp(this)`.

Reply:
Incredible!! I'm learning a lot with this! Will fix this by tomorrow.

Thanks a lot for the insight!

Reply:
I think this is fixed now.